ArduPlane/mode_guided.cpp

```diff
@@ -116,6 +127,33 @@ bool ModeGuided::handle_guided_request(Location target_loc)
 
     plane.set_guided_WP(target_loc);
 
+    // use waypoint navigation sub-mode
+    _guided_mode = SubMode::WP;
+
+    return true;
+}
+
+bool ModeGuided::handle_guided_path_request(Location position_on_path, Vector2f unit_path_tangent, const float path_curvature, const bool direction_is_ccw)
+{
+    // add home alt if needed
+    if (position_on_path.relative_alt) {
```

Comment:
There are various altitude types so it might be best to instead use the Location class's method to change the altitude type.